pkg/cortex/modules.go

```diff
@@ -614,3 +637,25 @@ var modules = map[ModuleName]module{
 		deps: []ModuleName{Querier, Ingester, Distributor, TableManager, DataPurger, StoreGateway},
 	},
 }
+
+// middleware for setting cache gen header to let consumer of response know all previous responses could be invalid due to delete operation
+func getHTTPCacheGenNumberHeaderSetterMiddleware(cacheGenNumbersLoader *purger.TombstonesLoader) middleware.Interface {
```

Comment:
If this is going to be an http middleware, it should live in either the `api` package or a function specific package.

Reply:
I added it to modules to keep `api` package purely for defining apis. Will see where I can move it. Thanks!

Reply:
The api package near the RegisterQuerier code would be good,  and then you can pass the tombstone loader directly into the registration function.

The other option would be adding a `CacheGenNumLoader` interface as a field in the API struct and when the tombstone loader is first instantiated, have it register with the API struct.